### ISSUE TITLE
Backup System

### ISSUE THREAD

**Issue by Micky007** [kiwibrowser/android#4]
Could you please add as a new feature a backup/restore system?
I would like it to mainly backup opened tabs, web navigation history, bookmarks and chrome flags and maybe settings.
The possibility to choose what part of the backup may be restored and to view its content would be useful.
The  backup should be made by creating a file (I guess it should be a text type) and nothing more.

At anyway your speed hacks crazy, my best compliments because this is the snappiest chromium build I've tried since maybe four years of various experiments with other chromium and non-chromium browsers.

**Comment by NotesOfReality** [comment]
Thank you very much for the precious information, if only I had known a bit of programming languages I would have implemented the backup system by myself and then created a pull request.

**Comment by gdgsdg123** [comment]
Just manually back them up/restore with any adequate explorer, actually...

**Comment by NotesOfReality** [comment]
Yeah, I already know that, I simply think that adding such a feature with a proper UI could make the app go in the top 5 or something like that

**Comment by madranet** [comment]
Thanks. This is good for helping back up an existing Kiwi config and restoring it. But doesn't address importing data from another browser. 

Please don't fall into the trap so many developers do of bringing out a new product and expecting people to migrate to it without being able to bring their existing data over from your rivals, who they've been [more or less] happily using for years.

Doesn't directly affect me as I have my bookmarks stored on pinboard. But a lot of folks will have years worth of bookmarks and history stored in `$CurrentBrowser` and are much more likely to move to `$NewBrowser` if they can bring that stuff with them.

**Comment by NotesOfReality** [comment]
> Thanks. This is good for helping back up an existing Kiwi config and restoring it. But doesn't address importing data from another browser.
> 
> Please don't fall into the trap so many developers do of bringing out a new product and expecting people to migrate to it without being able to bring their existing data over from your rivals, who they've been [more or less] happily using for years.
> 
> Doesn't directly affect me as I have my bookmarks stored on pinboard. But a lot of folks will have years worth of bookmarks and history stored in `$CurrentBrowser` and are much more likely to move to `$NewBrowser` if they can bring that stuff with them.

I think this is a quite worth of attention denotation, it now is so more than ever since Kiwi Browser developer even has brought Chrome extensions support.

**Comment by NotesOfReality** [comment]
In case anybody may ever need it, here I leave the "opened tabs" path, which has recently been tested by me.

opened tabs:
'/data/data/com.kiwibrowser.browser/app_tabs/'
copy the whole folder and replace the app_tabs/0/tab_state0 file or edit it adding the URLs listed there


Moreover I'd say we don't need the bookmarks path since we have the im/export feature.

**Comment by gdgsdg123** [comment]
> Moreover I'd say we don't need the bookmarks path since we have the bookmark im/export feature.

Prefer doing so by modifying the files anyway...
